### PR TITLE
[Do not merge yet] Rename `rustc-guide` to `rustc-dev-guide`

### PR DESCRIPTION
**Pull request by LeSeulArtichaut**
See rust-lang/rustc-guide#602.

### CODE REVIEW

**Comment by emilyalbini** [comment]
Ping me again when you need this to be merged.

File: rustc-dev-guide/index.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Redirecting to https://rust-lang.github.io/rustc-dev-guide/</title>
+<meta http-equiv="refresh" content="0; URL=https://rust-lang.github.io/rustc-dev-guide/">
```
Comment (chrissimpkins) [inline rustc-dev-guide/index.html]: Link updated to use https://rustc-dev-guide.rust-lang.org/ instead

```suggestion
<meta http-equiv="refresh" content="0; URL=https://rustc-dev-guide.rust-lang.org/">
```

File: rustc-dev-guide/index.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Redirecting to https://rust-lang.github.io/rustc-dev-guide/</title>
```
Comment (chrissimpkins) [inline rustc-dev-guide/index.html]:
```suggestion
<title>Redirecting to https://rustc-dev-guide.rust-lang.org/</title>
```

Comment (chrissimpkins) [reply rustc-dev-guide/index.html]: Use URL https://rustc-dev-guide.rust-lang.org/ instead

**Review by chrissimpkins — CHANGES_REQUESTED** [review]
We just need the URL updated here when you have a chance @LeSeulArtichaut.  Thanks!

**Review by mark-i-m — APPROVED** [review]

**Review by mark-i-m — CHANGES_REQUESTED** [review]
Hmm... actually, I think we want the old file name with the redirect URL updated.

**Comment by LeSeulArtichaut** [comment]
Err... you're right 😅

**Comment by LeSeulArtichaut** [comment]
Should be good to go now...

**Review by mark-i-m — APPROVED** [review]
Thanks! LGTM

Ping @pietroalbini

**Comment by chrissimpkins** [comment]
apologies!

**Pull request merged**